homeassistant/components/spaceapi.py

```diff
@@ -0,0 +1,171 @@
+        vol.Required(CONF_LOCATION): LOCATION_SCHEMA,
+        vol.Required(CONF_LOGO): cv.string,
+        vol.Required(CONF_SPACE): cv.string,
+        vol.Required(CONF_STATE): STATE_SCHEMA,
+        vol.Required(CONF_URL): cv.string,
+        vol.Optional(CONF_SENSORS): SENSOR_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Register the SpaceAPI with the HTTP interface."""
+    hass.data[DATA_SPACEAPI] = config[DOMAIN]
+    hass.http.register_view(APISpaceApiView)
+
+    return True
+
+
+class APISpaceApiView(HomeAssistantView):
+    """View to provide details according to the SpaceAPI."""
+
+    url = URL_API_SPACEAPI
+    name = 'api:spaceapi'
+
+    @ha.callback
+    def get(self, request):
+        """Get SpaceAPI data."""
+        spaceapi = dict(request.app['hass'].data[DATA_SPACEAPI])
+        is_sensors = spaceapi.get('sensors')
+
+        location = {
+            ATTR_ADDRESS: spaceapi[ATTR_LOCATION][CONF_ADDRESS],
+            ATTR_LATITUDE: request.app['hass'].config.latitude,
+            ATTR_LONGITUDE: request.app['hass'].config.longitude,
+        }
+
+        state_entity = spaceapi['state'][ATTR_ENTITY_ID]
+        state = request.app['hass'].states.get(state_entity)
+        last_changed = request.app['hass'].states.get(
+            state_entity).last_updated
+
+        state = {
+            ATTR_ICON: {
+                ATTR_OPEN: spaceapi['state'][CONF_ICON_OPEN],
+                ATTR_CLOSE: spaceapi['state'][CONF_ICON_CLOSED],
+            },
+            ATTR_OPEN: bool(state),
+            ATTR_LASTCHANGE: dt_util.as_timestamp(last_changed),
+        }
+
+        if is_sensors is not None:
+            sensors = {}
+            for sensor_type in SENSOR_TYPES:
+                sensors[sensor_type] = []
+                for sensor in spaceapi['sensors'][sensor_type]:
+                    sensor_state = request.app['hass'].states.get(sensor)
+                    name = sensor_state.attributes[ATTR_FRIENDLY_NAME]
+                    unit = sensor_state.attributes[ATTR_UNIT_OF_MEASUREMENT]
+                    value = sensor_state.state
+                    sensor_data = {
+                        ATTR_LOCATION: spaceapi[CONF_SPACE],
+                        ATTR_NAME: sensor if name is None else name,
+                        ATTR_UNIT: unit,
+                        ATTR_VALUE: value,
+                    }
+                    sensors[sensor_type].append(sensor_data)
+
+        data = {
+            ATTR_API: SPACEAPI_VERSION,
+            ATTR_CONTACT: spaceapi[CONF_CONTACT],
+            ATTR_ISSUE_REPORT_CHANNELS: spaceapi[CONF_ISSUE_REPORT_CHANNELS],
+            ATTR_LOCATION: location,
+            ATTR_LOGO: spaceapi[CONF_LOGO],
+            ATTR_SPACE: spaceapi[CONF_SPACE],
+            ATTR_STATE: state,
+            ATTR_URL: spaceapi[CONF_URL],
+        }
+
+        if is_sensors is not None:
+            data[ATTR_SENSORS] = sensors
```

Comment:
If you move the whole `data = {}` initializer up a few lines (above the first `if is_sensors is not None`) this second check could be avoided.

Reply:
Moved